pkg/client/record/event.go

```diff
@@ -44,50 +46,94 @@ type EventSink interface {
 	Update(event *api.Event) (*api.Event, error)
 }
 
-var emptySource = api.EventSource{}
+// EventRecorder knows how to record events on behalf of an EventSource.
+type EventRecorder interface {
+	// Event constructs an event from the given information and puts it in the queue for sending.
+	// 'object' is the object this event is about. Event will make a reference-- or you may also
+	// pass a reference to the object directly.
+	// 'reason' is the reason this event is generated. 'reason' should be short and unique; it will
+	// be used to automate handling of events, so imagine people writing switch statements to
+	// handle them. You want to make that easy.
+	// 'message' is intended to be human readable.
+	//
+	// The resulting event will be created in the same namespace as the reference object.
+	Event(object runtime.Object, reason, message string)
+
+	// Eventf is just like Event, but with Sprintf for the message field.
+	Eventf(object runtime.Object, reason, messageFmt string, args ...interface{})
+}
+
+// EventBroadcaster knows how to recieve events and send them to various sources.
+type EventBroadcaster interface {
+	// StartEventWatcher starts sending events recieved from this EventBroadcaster to the given
+	// event handler function. The return value can be ignored or used to stop recording, if
+	// desired.
+	StartEventWatcher(eventHandler func(*api.Event)) watch.Interface
+
+	// StartRecordingToSink starts sending events recieved from this EventBroadcaster to the given
+	// sink. The return value can be ignored or used to stop recording, if desired.
+	StartRecordingToSink(sink EventSink) watch.Interface
+
+	// StartLogging starts sending events recieved from this EventBroadcaster to the given logging
+	// function. The return value can be ignored or used to stop recording, if desired.
+	StartLogging(logf func(format string, args ...interface{})) watch.Interface
+
+	// NewRecorder returns an EventRecorder that can be used to send events to this EventBroadcaster
+	// with the event source set to the given event source.
+	NewRecorder(source api.EventSource) EventRecorder
```

Comment:
OK-- I don't have a super strong opinion, but creating one broadcaster per server object (instead of per binary) just seems a little inconsistent. Like presumably one server object won't ever want to send events from more than one source.